modules/post/windows/gather/hashdump.rb

```diff
@@ -241,16 +251,37 @@ def rid_to_key(rid)
     [convert_des_56_to_64(s1), convert_des_56_to_64(s2)]
   end
 
-  def decrypt_user_hash(rid, hbootkey, enchash, pass)
+  def decrypt_user_hash(rid, hbootkey, enchash, pass, default)
+    revision = enchash[2, 2].unpack('v')[0]
 
-    if(enchash.empty?)
-      case pass
-      when @sam_lmpass
-        return @sam_empty_lm
-      when @sam_ntpass
-        return @sam_empty_nt
+    if revision == 1
+      if enchash.length < 20
+        return default
       end
-      return ""
+
+      md5 = Digest::MD5.new
+      md5.update(hbootkey[0,16] + [rid].pack("V") + pass)
+
+      rc4 = OpenSSL::Cipher.new('rc4')
+      rc4.key = md5.digest
+      okey = rc4.update(enchash[4, 16])
+
+    elsif revision == 2
```

Comment:
This could also be a case statement, but there's nothing actually wrong with the way it is now. I just think case statements are a but cleaner under these circumstances, and allow for easier future expansions if needed